api/common_test.go

```diff
@@ -0,0 +1,207 @@
+			PrivatePort: 9998,
+			Type:        "udp",
+		}, {
+			IP:          "1.2.3.4",
+			PublicPort:  9999,
+			PrivatePort: 9999,
+			Type:        "udp",
+		}},
+			"1.2.3.4:9998-9999->9998-9999/udp",
+		},
+		{[]types.Port{{
+			IP:          "1.2.3.4",
+			PublicPort:  8887,
+			PrivatePort: 9998,
+			Type:        "udp",
+		}, {
+			IP:          "1.2.3.4",
+			PublicPort:  8888,
+			PrivatePort: 9999,
+			Type:        "udp",
+		}},
+			"1.2.3.4:8887->9998/udp, 1.2.3.4:8888->9999/udp",
+		},
+		{[]types.Port{{
+			PrivatePort: 9998,
+			Type:        "udp",
+		}, {
+			PrivatePort: 9999,
+			Type:        "udp",
+		}},
+			"9998-9999/udp",
+		},
+		{[]types.Port{{
+			IP:          "1.2.3.4",
+			PrivatePort: 6677,
+			PublicPort:  7766,
+			Type:        "tcp",
+		}, {
+			PrivatePort: 9988,
+			PublicPort:  8899,
+			Type:        "udp",
+		}},
+			"9988/udp, 1.2.3.4:7766->6677/tcp",
+		},
+		{[]types.Port{{
+			IP:          "1.2.3.4",
+			PrivatePort: 9988,
+			PublicPort:  8899,
+			Type:        "udp",
+		}, {
+			IP:          "1.2.3.4",
+			PrivatePort: 9988,
+			PublicPort:  8899,
+			Type:        "tcp",
+		}, {
+			IP:          "4.3.2.1",
+			PrivatePort: 2233,
+			PublicPort:  3322,
+			Type:        "tcp",
+		}},
+			"4.3.2.1:3322->2233/tcp, 1.2.3.4:8899->9988/udp, 1.2.3.4:8899->9988/tcp",
+		},
+		{[]types.Port{{
+			PrivatePort: 9988,
+			PublicPort:  8899,
+			Type:        "udp",
+		}, {
+			IP:          "1.2.3.4",
+			PrivatePort: 6677,
+			PublicPort:  7766,
+			Type:        "tcp",
+		}, {
+			IP:          "4.3.2.1",
+			PrivatePort: 2233,
+			PublicPort:  3322,
+			Type:        "tcp",
+		}},
+			"9988/udp, 4.3.2.1:3322->2233/tcp, 1.2.3.4:7766->6677/tcp",
+		},
+	}
+
+	for _, port := range cases {
+		actual := DisplayablePorts(port.ports)
+		if port.expected != actual {
+			t.Fatalf("Expected %s, got %s.", port.expected, actual)
+		}
+	}
+}
+
+// MatchesContentType
+func TestJsonContentType(t *testing.T) {
+	if !MatchesContentType("application/json", "application/json") {
+		t.Fail()
+	}
+
+	if !MatchesContentType("application/json; charset=utf-8", "application/json") {
+		t.Fail()
+	}
+
+	if MatchesContentType("dockerapplication/json", "application/json") {
+		t.Fail()
+	}
+}
+
+// LoadOrCreateTrustKey
+func TestLoadOrCreateTrustKeyInvalidKeyFile(t *testing.T) {
+	tmpKeyFolderPath, _ := ioutil.TempDir("", "api-trustkey-test")
+	defer os.RemoveAll(tmpKeyFolderPath)
+	tmpKeyFile, _ := ioutil.TempFile(tmpKeyFolderPath, "keyfile")
+
+	if _, err := LoadOrCreateTrustKey(tmpKeyFile.Name()); err == nil {
+		t.Fatalf("expected an error, got nothing. %v", err)
+	}
+
+}
+
+func TestLoadOrCreateTrustKeyCreateKey(t *testing.T) {
+	tmpKeyFolderPath, _ := ioutil.TempDir("", "api-trustkey-test")
+	defer os.RemoveAll(tmpKeyFolderPath)
+	tmpKeyFile := filepath.Join(tmpKeyFolderPath, "keyfile")
+
+	if key, err := LoadOrCreateTrustKey(tmpKeyFile); err != nil || key == nil {
+		t.Fatalf("expected a new key file, got : %v and %v", err, key)
+	}
+
```

Comment:
I think we should check to make sure the file exists on disk, since it possible that the Load... call could have returned valid info but messed up trying to save it to disk.  Don't think we need to check the contents of the file, rather just that the file exists and is non-empty.  Same for the next test too.